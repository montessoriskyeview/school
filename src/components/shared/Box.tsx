import { Box as MuiBox, BoxProps as MuiBoxProps } from '@mui/material';
import { forwardRef } from 'react';

// Simplified BoxProps that reduces union type complexity
export interface BoxProps extends Omit<MuiBoxProps, 'component'> {
  component?: React.ElementType;
}

// Custom Box component that defaults to div to avoid TypeScript union type complexity
export const Box = forwardRef<HTMLDivElement, BoxProps>(({ ...props }, ref) => {
  return <MuiBox ref={ref} {...props} component={'div'} />;
});

Box.displayName = 'Box';
