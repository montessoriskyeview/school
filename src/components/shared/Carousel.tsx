import React, { useState } from 'react';

interface Slide {
  src: string;
  className?: string;
}

export interface CarouselProps {
  slides: Slide[];
}

export const Carousel: React.FC<CarouselProps> = ({ slides }) => {
  const [currentIndex, setCurrentIndex] = useState(0);

  const nextSlide = () => {
    setCurrentIndex(prevIndex =>
      prevIndex === slides.length - 1 ? 0 : prevIndex + 1
    );
  };

  const prevSlide = () => {
    setCurrentIndex(prevIndex =>
      prevIndex === 0 ? slides.length - 1 : prevIndex - 1
    );
  };

  return (
    <div className='carousel-container'>
      <div className='carousel-content'>
        {slides.map((slide, index) => (
          <img
            key={index}
            src={slide.src}
            alt={`Slide ${index + 1}`}
            className={`carousel-slide ${slide.className || ''}`}
            style={{
              opacity: index === currentIndex ? 1 : 0,
              transform: `translateX(${(index - currentIndex) * 100}%)`
            }}
          />
        ))}
      </div>

      <div className='navigation-controls'>
        <button className='carousel-button prev' onClick={prevSlide}>
          ❮
        </button>
        <div className='carousel-counter'>
          {currentIndex + 1}/{slides.length}
        </div>
        <button className='carousel-button next' onClick={nextSlide}>
          ❯
        </button>
      </div>

      <style>{`
        .carousel-container {
          position: relative;
          width: 100%;
          height: 500px;
          max-width: 800px;
          margin: 0 auto;
          overflow: hidden;
        }

        .carousel-content {
          position: relative;
          height: 100%;
          width: 100%;
        }

        .carousel-slide {
          position: absolute;
          top: 0;
          left: 0;
          width: 100%;
          height: 100%;
          object-fit: cover;
          transition: all 0.5s ease-in-out;
        }

        .navigation-controls {
          position: absolute;
          bottom: 20px;
          left: 50%;
          transform: translateX(-50%);
          display: flex;
          align-items: center;
          gap: 16px;
          background: rgba(0, 0, 0, 0.3);
          padding: 8px 16px;
          border-radius: 20px;
        }

        .carousel-button {
          position: static;
          transform: none;
          padding: 8px 12px;
          border-radius: 50%;
          background: rgba(255, 255, 255, 0.2);
          color: white;
          border: none;
          cursor: pointer;
          font-size: 14px;
          transition: background 0.3s ease;
        }

        .carousel-button:hover {
          background: rgba(255, 255, 255, 0.4);
        }

        .carousel-counter {
          color: white;
          font-size: 14px;
          min-width: 45px;
          text-align: center;
        }
      `}</style>
    </div>
  );
};

/**
 * 
 * 
const itemData = [
  {
    img: property1,
    rows: 2,
    cols: 2
  },
  {
    img: property2
  },
  {
    img: property3
  },
  {
    img: property4,
    cols: 2
  },
  {
    img: property5,
    cols: 2
  },
  {
    img: property6,
    rows: 2,
    cols: 2
  },
  {
    img: property7
  },
  {
    img: property8
  },
  {
    img: property9,
    rows: 2,
    cols: 2
  },
  {
    img: property10
  },
  {
    img: property11
  },
  {
    img: property12,
    cols: 2
  },
  {
    img: property13
  },
  {
    img: property14
  },
  {
    img: property15
  },
  {
    img: property16
  }
];

function srcset(image: string, size: number, rows = 1, cols = 1) {
  return {
    src: `${image}?w=${size * cols}&h=${size * rows}&fit=crop&auto=format`,
    srcSet: `${image}?w=${size * cols}&h=${
      size * rows
    }&fit=crop&auto=format&dpr=2 2x`
  };
}

 * 
 */
