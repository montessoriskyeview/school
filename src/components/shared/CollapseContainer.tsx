import { Box, Button, Collapse } from "@mui/material";
import { ContentContainer } from "./ContentContainer"
import { useState } from "react";
import ExpandLessIcon from '@mui/icons-material/ExpandLess';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import { Typography } from "./Typography";

export const CollapseContainer = ({ title, content }: { title: string; content: string | JSX.Element }) => {
  const [open, setOpen] = useState(false);

  const handleClick = () => {
    setOpen(o => !o);
  };

  return (
    <ContentContainer>
      <Box component="div" sx={{ width: "100%" }}>
        <Button variant="text" sx={{ width: "100%", justifyContent: "left", gap: 3, textTransform: "none", textAlign: "left" }} onClick={handleClick}>
          {open ? <ExpandLessIcon /> : <ExpandMoreIcon />}
          <Typography sx={{ color: "white" }}>{title}</Typography>
        </Button>
      </Box>
      <Collapse in={open}>
        <Box component="div" sx={{ paddingTop: "10px" }}>
          {content}
        </Box>
      </Collapse>
    </ContentContainer>
  )
}