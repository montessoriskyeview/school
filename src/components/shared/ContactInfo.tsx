import { ContentContainer } from "./ContentContainer"
import { Typography } from "./Typography"

const PHONE = "7605340671"
const EMAIL = "NWLVMontessori@gmail.com"

export const ContactInfo = () => {
  return (
    <ContentContainer>
      <>
        <Typography>
          {"For questions or concerns, contact Hannah Olsen (call/text/email)"}
        </Typography>
        <Typography>
          <a href={`tel:${PHONE}`}>{"(760) 534-0671"}</a>
        </Typography>
        <Typography>
          <a href={`mailto:${EMAIL}`}>{EMAIL}</a>
        </Typography>
      </>
    </ContentContainer >
  )
}