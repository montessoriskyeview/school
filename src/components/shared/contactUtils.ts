/**
 * Contact utilities for Montessori Skye View Learning Center
 *
 * This file contains shared contact information and utility functions
 * to ensure consistency across all components that handle contact interactions.
 */

// Contact Information Constants
export const PHONE = '7605340671';
export const PHONE_LINK = `tel:${PHONE}`;
export const PHONE_FORMATTED = '(760) 534-0671';
export const PHONE_INTERNATIONAL = '+1-760-534-0671';
export const EMAIL = 'NWLVMontessori@gmail.com'.toLowerCase();

// Email Template Configuration
export const EMAIL_SUBJECT = 'Inquiry from Montessori Skye View Website';
export const INCLUDE_BODY = false;
export const EMAIL_BODY = `Hello,

I'm interested in learning more about Montessori Skye View Learning Center.

Please provide me with information about:
- Tour availability
- Enrollment availability
- Program details
- Tuition information
- Any other relevant information

Thank you!`;

/**
 * Creates a mailto URL with consistent subject and body for all email links
 *
 * This function ensures that all email links across the website use the same
 * subject line and body template, providing a consistent experience for users
 * and making it easier for staff to identify and respond to website inquiries.
 *
 * @returns {string} Complete mailto URL with subject and body parameters
 *
 * @example
 * ```tsx
 * <a href={createEmailLink()}>Email Us</a>
 * ```
 */
export const createEmailLink = (): string => {
  const encodedSubject = encodeURIComponent(EMAIL_SUBJECT);
  const encodedBody = encodeURIComponent(EMAIL_BODY);

  if (INCLUDE_BODY) {
    return `mailto:${EMAIL}?subject=${encodedSubject}&body=${encodedBody}`;
  }

  return `mailto:${EMAIL}?subject=${encodedSubject}`;
};

/**
 * Creates a phone link with the standard phone number
 *
 * @returns {string} Complete tel URL
 */
export const createPhoneLink = (): string => {
  return `tel:${PHONE}`;
};
