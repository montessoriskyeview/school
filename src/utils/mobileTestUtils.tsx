import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

// Mobile viewport configurations
export const MOBILE_VIEWPORTS = {
  iPhoneSE: { width: 375, height: 667 },
  iPhone12: { width: 390, height: 844 },
  iPhone12ProMax: { width: 428, height: 926 },
  SamsungGalaxyS20: { width: 360, height: 800 },
  iPad: { width: 768, height: 1024 },
};

// Set mobile viewport for testing
export const setMobileViewport = (
  viewport: keyof typeof MOBILE_VIEWPORTS = 'iPhoneSE'
) => {
  const { width, height } = MOBILE_VIEWPORTS[viewport];

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  Object.defineProperty(window, 'innerHeight', {
    writable: true,
    configurable: true,
    value: height,
  });

  // Trigger resize event
  window.dispatchEvent(new Event('resize'));
};

// Custom render function with mobile context
const AllTheProviders: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  return <BrowserRouter>{children}</BrowserRouter>;
};

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// Mobile-specific test helpers
export const mobileTestUtils = {
  // Test touch target size
  expectTouchTarget: (element: HTMLElement, minSize = 44) => {
    const rect = element.getBoundingClientRect();
    expect(rect.width).toBeGreaterThanOrEqual(minSize);
    expect(rect.height).toBeGreaterThanOrEqual(minSize);
  },

  // Test mobile spacing
  expectMobileSpacing: (element: HTMLElement, minSpacing = 24) => {
    const computedStyle = window.getComputedStyle(element);
    const padding =
      parseFloat(computedStyle.paddingTop) +
      parseFloat(computedStyle.paddingBottom);
    const margin =
      parseFloat(computedStyle.marginTop) +
      parseFloat(computedStyle.marginBottom);
    const totalSpacing = padding + margin;
    expect(totalSpacing).toBeGreaterThanOrEqual(minSpacing);
  },

  // Test mobile accessibility
  expectMobileAccessibility: (element: HTMLElement) => {
    const hasRole =
      element.getAttribute('role') || element.tagName.toLowerCase();
    const hasLabel =
      element.getAttribute('aria-label') ||
      element.getAttribute('alt') ||
      element.textContent?.trim();

    expect(hasRole).toBeTruthy();
    expect(hasLabel).toBeTruthy();
  },

  // Test mobile responsive behavior
  expectMobileResponsive: (element: HTMLElement) => {
    const computedStyle = window.getComputedStyle(element);
    const fontSize = parseFloat(computedStyle.fontSize);
    const lineHeight = parseFloat(computedStyle.lineHeight);

    // Check minimum font size for mobile
    expect(fontSize).toBeGreaterThanOrEqual(16);

    // Check minimum line height for readability
    expect(lineHeight).toBeGreaterThanOrEqual(1.4);
  },

  // Test mobile navigation
  expectMobileNavigation: (element: HTMLElement) => {
    const isClickable =
      element.tagName === 'BUTTON' ||
      element.tagName === 'A' ||
      element.getAttribute('role') === 'button' ||
      element.onclick !== null;

    expect(isClickable).toBe(true);
  },
};

// Export custom render
export { customRender as render };

// Re-export everything from testing library
export * from '@testing-library/react';
