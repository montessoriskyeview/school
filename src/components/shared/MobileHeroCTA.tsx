import {
  Box,
  Container,
  Button,
  Stack,
  IconButton,
  Collapse,
} from '@mui/material';
import { EnrollmentButtons } from './EnrollmentButtons';
import { useState } from 'react';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import ExpandLessIcon from '@mui/icons-material/ExpandLess';
import { createEmailLink, createPhoneLink } from './contactUtils';

export const MobileHeroCTA = () => {
  const [expanded, setExpanded] = useState(false);

  const toggleExpanded = () => {
    setExpanded(!expanded);
  };

  return (
    <Box
      component="div"
      sx={{
        display: { xs: 'block', md: 'none' }, // Only show on mobile
        position: 'fixed', // Changed from sticky to fixed
        bottom: -2,
        left: 0,
        right: 0,
        zIndex: 1400, // Higher than AppBar and most other components
        width: '100%',
        backgroundColor: 'var(--white)',
        borderRadius: { xs: 0, sm: 'var(--radius-lg) var(--radius-lg) 0 0' }, // No radius on very small screens
        boxShadow: '0 -4px 20px rgba(0,0,0,0.15)', // Shadow pointing upward
        border: '2px solid var(--primary-blue)',
        borderBottom: 'none', // Remove bottom border since it's at screen edge
        overflow: 'hidden',
        // Add safe area padding for devices with notches/home indicators
        paddingBottom: 'env(safe-area-inset-bottom)',
      }}
    >
      {/* Collapsible Header */}
      <Box
        component="div"
        sx={{
          padding: { xs: 1.5, md: 0 },
          backgroundColor: 'var(--primary-blue)',
          color: 'var(--white)',
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'space-between',
          cursor: 'pointer',
          minHeight: 48, // Ensure touch target
          '&:hover': {
            backgroundColor: '#1D4ED8',
          },
        }}
        onClick={toggleExpanded}
        role="button"
        tabIndex={0}
        aria-expanded={expanded}
        aria-label={
          expanded ? 'Collapse enrollment options' : 'Expand enrollment options'
        }
        onKeyDown={(e: React.KeyboardEvent) => {
          if (e.key === 'Enter' || e.key === ' ') {
            e.preventDefault();
            toggleExpanded();
          }
        }}
      >
        <Box
          component="div"
          sx={{
            display: 'flex',
            alignItems: 'center',
            gap: 1,
          }}
        >
          <span
            style={{
              fontSize: 'var(--text-base)',
              fontWeight: 600,
            }}
          >
            📚 Ready to Enroll?
          </span>
        </Box>
        <IconButton
          size="small"
          sx={{
            color: 'var(--white)',
            padding: 0.5,
            '&:hover': {
              backgroundColor: 'rgba(255, 255, 255, 0.1)',
            },
          }}
          aria-hidden="true"
        >
          {expanded ? <ExpandLessIcon /> : <ExpandMoreIcon />}
        </IconButton>
      </Box>

      {/* Collapsible Content */}
      <Collapse in={expanded} timeout="auto" unmountOnExit>
        <Container
          maxWidth="sm"
          sx={{
            textAlign: 'center',
            py: { xs: 2, md: 0 },
            px: { xs: 2, md: 0 },
          }}
        >
          <Box
            component="div"
            sx={{
              marginBottom: 'var(--spacing-lg)',
            }}
          >
            <p
              style={{
                fontSize: 'var(--text-base)',
                color: 'var(--text-secondary)',
                margin: '0 0 var(--spacing-lg) 0',
                lineHeight: 1.5,
              }}
            >
              Join our Montessori community today
            </p>
          </Box>

          <EnrollmentButtons
            variant="primary"
            sx={{
              flexDirection: 'column',
              gap: 2,
              '& .MuiLink-root': {
                fontSize: 'var(--text-base)',
                fontWeight: 700,
                padding: 'var(--spacing-md) var(--spacing-lg)',
                borderRadius: 'var(--radius-full)',
                backgroundColor: 'var(--primary-blue)',
                color: 'var(--white)',
                textDecoration: 'none',
                display: 'inline-block',
                transition: 'all 0.3s ease',
                boxShadow: 'var(--shadow-md)',
                minHeight: '44px',
                '&:hover': {
                  backgroundColor: '#1D4ED8',
                  transform: 'translateY(-2px)',
                  boxShadow: 'var(--shadow-lg)',
                  textDecoration: 'none',
                },
                '&:focus': {
                  outline: '3px solid var(--primary-blue)',
                  outlineOffset: '2px',
                },
              },
            }}
          />

          {/* Contact Buttons */}
          <Stack
            direction="row"
            spacing={2}
            justifyContent="center"
            sx={{ mt: 2 }}
          >
            <Button
              variant="outlined"
              color="primary"
              href={createPhoneLink()}
              sx={{
                minWidth: 0,
                px: 2,
                py: 1,
                fontSize: 'var(--text-base)',
                borderRadius: 'var(--radius-full)',
                textTransform: 'none',
                fontWeight: 600,
                lineHeight: 1.2,
                minHeight: 44, // Ensure touch target
              }}
              aria-label="Call Montessori Skye View"
            >
              📞 Call
            </Button>
            <Button
              variant="outlined"
              color="primary"
              href={createEmailLink()}
              sx={{
                minWidth: 0,
                px: 2,
                py: 1,
                fontSize: 'var(--text-base)',
                borderRadius: 'var(--radius-full)',
                textTransform: 'none',
                fontWeight: 600,
                lineHeight: 1.2,
                minHeight: 44, // Ensure touch target
              }}
              aria-label="Email Montessori Skye View"
            >
              ✉️ Email
            </Button>
          </Stack>
        </Container>
      </Collapse>
    </Box>
  );
};
