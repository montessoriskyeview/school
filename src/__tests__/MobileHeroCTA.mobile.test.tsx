import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { setMobileViewport } from '../utils/mobileTestUtils';
import { MobileHeroCTA } from '../components/shared/MobileHeroCTA';

// Mock the new contact components
jest.mock('../components/shared/EmailContact', () => ({
  EmailContact: ({ children, ...props }: any) => (
    <a href="mailto:test@example.com" aria-label="Email Montessori Skye View" {...props}>
      {children}
    </a>
  ),
}));

jest.mock('../components/shared/PhoneContact', () => ({
  PhoneContact: ({ children, ...props }: any) => (
    <a href="tel:+1234567890" aria-label="Call Montessori Skye View" {...props}>
      {children}
    </a>
  ),
}));

describe('MobileHeroCTA Component - Mobile Experience', () => {
  beforeEach(() => {
    // Set mobile viewport for all tests
    setMobileViewport('iPhoneSE');
  });

  describe('Mobile Layout and Positioning', () => {
    test('renders correctly on mobile viewport', () => {
      render(<MobileHeroCTA />);
      expect(
        screen.getByRole('button', { name: /expand enrollment options/i })
      ).toBeInTheDocument();
    });

    test('component is visible and accessible', () => {
      render(<MobileHeroCTA />);
      const header = screen.getByRole('button', {
        name: /expand enrollment options/i,
      });
      expect(header).toBeVisible();
      expect(header).toBeInTheDocument();
    });

    test('snapshot matches mobile layout', () => {
      const { container } = render(<MobileHeroCTA />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('Mobile Touch Interactions', () => {
    test('expand/collapse functionality works with touch', async () => {
      render(<MobileHeroCTA />);

      const header = screen.getByRole('button', {
        name: /expand enrollment options/i,
      });

      // Initially collapsed
      expect(header).toHaveAttribute('aria-expanded', 'false');

      // Click to expand
      fireEvent.click(header);

      await waitFor(() => {
        expect(header).toHaveAttribute('aria-expanded', 'true');
      });

      // Click to collapse
      fireEvent.click(header);

      await waitFor(() => {
        expect(header).toHaveAttribute('aria-expanded', 'false');
      });
    });

    test('keyboard navigation works for accessibility', async () => {
      render(<MobileHeroCTA />);

      const header = screen.getByRole('button', {
        name: /expand enrollment options/i,
      });

      // Test Enter key
      fireEvent.keyDown(header, { key: 'Enter' });

      await waitFor(() => {
        expect(header).toHaveAttribute('aria-expanded', 'true');
      });

      // Test Space key
      fireEvent.keyDown(header, { key: ' ' });

      await waitFor(() => {
        expect(header).toHaveAttribute('aria-expanded', 'false');
      });
    });
  });

  describe('Mobile Content and Typography', () => {
    test('displays correct enrollment message', () => {
      render(<MobileHeroCTA />);

      expect(screen.getByText(/ready to enroll/i)).toBeInTheDocument();
    });

    test('enrollment buttons are present when expanded', async () => {
      render(<MobileHeroCTA />);

      // Expand the component first
      const header = screen.getByRole('button', {
        name: /expand enrollment options/i,
      });
      fireEvent.click(header);

      await waitFor(() => {
        const enrollmentButtons = screen.getAllByRole('link');
        expect(enrollmentButtons.length).toBeGreaterThan(0);
      });
    });

    test('contact buttons are present when expanded', async () => {
      render(<MobileHeroCTA />);

      // Expand the component first
      const header = screen.getByRole('button', {
        name: /expand enrollment options/i,
      });
      fireEvent.click(header);

      await waitFor(() => {
        const callButton = screen.getByRole('link', { name: /call/i });

        expect(callButton).toBeInTheDocument();
      });

      await waitFor(() => {
        const emailButton = screen.getByRole('link', { name: /email/i });

        expect(emailButton).toBeInTheDocument();
      });
    });
  });

  describe('Mobile Accessibility', () => {
    test('component has proper ARIA attributes', () => {
      render(<MobileHeroCTA />);

      const header = screen.getByRole('button', {
        name: /expand enrollment options/i,
      });

      expect(header).toHaveAttribute('aria-expanded');
      expect(header).toHaveAttribute('aria-label');
      expect(header).toHaveAttribute('tabIndex');
    });

    test('expand/collapse state is properly announced', async () => {
      render(<MobileHeroCTA />);

      const header = screen.getByRole('button', {
        name: /expand enrollment options/i,
      });

      // Check initial state
      expect(header).toHaveAttribute('aria-label', 'Expand enrollment options');

      // Expand and check updated state
      fireEvent.click(header);

      await waitFor(() => {
        expect(header).toHaveAttribute(
          'aria-label',
          'Collapse enrollment options'
        );
      });
    });

    test('contact buttons have proper ARIA labels', async () => {
      render(<MobileHeroCTA />);

      // Expand the component first
      const header = screen.getByRole('button', {
        name: /expand enrollment options/i,
      });
      fireEvent.click(header);

      await waitFor(() => {
        const callButton = screen.getByRole('link', { name: /call/i });

        expect(callButton).toHaveAttribute('aria-label');
      });

      await waitFor(() => {
        const emailButton = screen.getByRole('link', { name: /email/i });

        expect(emailButton).toHaveAttribute('aria-label');
      });
    });

    test('focus management works correctly', async () => {
      render(<MobileHeroCTA />);

      const header = screen.getByRole('button', {
        name: /expand enrollment options/i,
      });

      // Focus the header
      header.focus();
      expect(header).toHaveFocus();

      // Expand and check if focus remains
      fireEvent.click(header);

      await waitFor(() => {
        expect(header).toHaveFocus();
      });
    });
  });

  describe('Mobile Performance', () => {
    test('expands and collapses smoothly', async () => {
      const startTime = performance.now();

      render(<MobileHeroCTA />);

      const header = screen.getByRole('button', {
        name: /expand enrollment options/i,
      });
      fireEvent.click(header);

      await waitFor(() => {
        expect(header).toHaveAttribute('aria-expanded', 'true');
      });

      const endTime = performance.now();
      const expandTime = endTime - startTime;

      // Should expand within 100ms
      expect(expandTime).toBeLessThan(100);
    });

    test('no console errors during interactions', () => {
      const consoleSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(<MobileHeroCTA />);

      const header = screen.getByRole('button', {
        name: /expand enrollment options/i,
      });
      fireEvent.click(header);

      expect(consoleSpy).not.toHaveBeenCalled();
      consoleSpy.mockRestore();
    });
  });

  describe('Mobile Responsive Behavior', () => {
    test('renders correctly on different mobile viewports', () => {
      const viewports = ['iPhoneSE', 'iPhone12', 'SamsungGalaxyS20'] as const;

      viewports.forEach(viewport => {
        setMobileViewport(viewport);
        const { container } = render(<MobileHeroCTA />);
        expect(container).toMatchSnapshot();
      });
    });

    test('maintains functionality across mobile devices', () => {
      const viewports = ['iPhoneSE', 'iPhone12', 'SamsungGalaxyS20'] as const;

      viewports.forEach(viewport => {
        setMobileViewport(viewport);
        render(<MobileHeroCTA />);

        const headers = screen.getAllByRole('button', {
          name: /expand enrollment options/i,
        });
        expect(headers.length).toBeGreaterThan(0);
        expect(headers[0]).toBeInTheDocument();
      });
    });
  });

  describe('Mobile Content Validation', () => {
    test('contact buttons have correct links', async () => {
      render(<MobileHeroCTA />);

      // Expand the component first
      const header = screen.getByRole('button', {
        name: /expand enrollment options/i,
      });
      fireEvent.click(header);

      await waitFor(() => {
        const emailButton = screen.getByRole('link', { name: /email/i });

        expect(emailButton).toHaveAttribute('href', 'mailto:test@example.com');
      });

      await waitFor(() => {
        const callButton = screen.getByRole('link', { name: /call/i });

        expect(callButton).toHaveAttribute('href', 'tel:+1234567890');
      });
    });

    test('enrollment buttons have valid links', async () => {
      render(<MobileHeroCTA />);

      // Expand the component first
      const header = screen.getByRole('button', {
        name: /expand enrollment options/i,
      });
      fireEvent.click(header);

      await waitFor(() => {
        const enrollmentButtons = screen.getAllByRole('link');
        enrollmentButtons.forEach(button => {
          expect(button).toHaveAttribute('href');
          const href = button.getAttribute('href');
          // Check for valid link types (https, tel, mailto)
          expect(href).toMatch(/^(https:\/\/|tel:|mailto:)/);
        });
      });
    });
  });
});
