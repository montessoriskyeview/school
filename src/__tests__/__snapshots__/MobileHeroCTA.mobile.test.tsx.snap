// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileHeroCTA Component - Mobile Experience Mobile Layout and Positioning snapshot matches mobile layout 1`] = `
<div>
  <div
    class="MuiBox-root css-1jh62j7"
  >
    <div
      aria-expanded="false"
      aria-label="Expand enrollment options"
      class="MuiBox-root css-tpqms"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiBox-root css-axw7ok"
      >
        <span
          style="font-weight: 600;"
        >
          ðŸ“š Ready to Enroll?
        </span>
      </div>
      <button
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-smsurf-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`MobileHeroCTA Component - Mobile Experience Mobile Responsive Behavior renders correctly on different mobile viewports 1`] = `
<div>
  <div
    class="MuiBox-root css-1jh62j7"
  >
    <div
      aria-expanded="false"
      aria-label="Expand enrollment options"
      class="MuiBox-root css-tpqms"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiBox-root css-axw7ok"
      >
        <span
          style="font-weight: 600;"
        >
          ðŸ“š Ready to Enroll?
        </span>
      </div>
      <button
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-smsurf-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`MobileHeroCTA Component - Mobile Experience Mobile Responsive Behavior renders correctly on different mobile viewports 2`] = `
<div>
  <div
    class="MuiBox-root css-1jh62j7"
  >
    <div
      aria-expanded="false"
      aria-label="Expand enrollment options"
      class="MuiBox-root css-tpqms"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiBox-root css-axw7ok"
      >
        <span
          style="font-weight: 600;"
        >
          ðŸ“š Ready to Enroll?
        </span>
      </div>
      <button
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-smsurf-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`MobileHeroCTA Component - Mobile Experience Mobile Responsive Behavior renders correctly on different mobile viewports 3`] = `
<div>
  <div
    class="MuiBox-root css-1jh62j7"
  >
    <div
      aria-expanded="false"
      aria-label="Expand enrollment options"
      class="MuiBox-root css-tpqms"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiBox-root css-axw7ok"
      >
        <span
          style="font-weight: 600;"
        >
          ðŸ“š Ready to Enroll?
        </span>
      </div>
      <button
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-smsurf-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
